### PR TITLE
Improvement error handling

### PR DESCRIPTION
**Pull request by govind-basepair**
### NOT READY TO MEGRE
**Task:** https://app.asana.com/0/577079795655810/1183037375235361/f
**TODO:** 
Have to update the error, warning message display after updating the response in analysis API in webapp

### CODE REVIEW

File: basepair/api.py [hunk 1]
```diff
@@ -410,11 +426,15 @@ def create_analysis(self, workflow_id, sample_id=None, sample_ids=None,
...
         else:
-            print('failed:', ','.join(sample_ids),
-                  res.status_code, file=sys.stderr)
-
+            error_msgs = {
+                401: 'You don\'t have access to this resource.',
+                404: 'Resource not found.',
+                500: 'Error retrieving data from API!'
+            }
+            if res.status_code in error_msgs:
```
Comment (saurabhbp) [inline basepair/api.py:435]: any 4** or 5** code should be considered an error. you can show some generic error message for the others. if error message or warning message list exists, we should show the list of errors formatted well for reading on cli. we can print the json if the json flag is enabled.

Comment (saurabhbp) [reply basepair/api.py:435]: can have:
```
Warning(s):
- warning 1
- warning 2
Error(s):
- error 1
- error 2
```

Comment (saurabhbp) [reply basepair/api.py:435]: if the response object doesn't have this format, we can default to printing the response text directly.

Comment (govind-basepair) [reply basepair/api.py:435]: Yes. These changes have to be done once we update the error response structure of APIs in the backend. A task is added for this - https://app.asana.com/0/577079795655810/1183927454734641/f

Comment (saurabhbp) [reply basepair/api.py:435]: cool

**Pull request merged**